File: app/src/main/java/com/worldventures/dreamtrips/wallet/ui/settings/help/feedback/payment/PaymentFeedbackScreen.java
package com.worldventures.dreamtrips.wallet.ui.settings.help.feedback.payment;

import com.worldventures.dreamtrips.wallet.ui.settings.help.feedback.base.BaseFeedbackScreen;
import com.worldventures.dreamtrips.wallet.ui.settings.help.feedback.payment.model.PaymentFeedbackViewModel;

import rx.Observable;

public interface PaymentFeedbackScreen extends BaseFeedbackScreen {

   Observable<CharSequence> observeMerchantName();

   PaymentFeedbackViewModel getPaymentFeedbackViewModel();

   void showBackConfirmDialog();

   void discardViewModelChanges();
}

File: wallet/src/main/java/com/worldventures/wallet/ui/settings/help/feedback/payment/PaymentFeedbackScreen.kt
package com.worldventures.wallet.ui.settings.help.feedback.payment

import com.worldventures.wallet.ui.settings.help.feedback.base.BaseFeedbackScreen
import com.worldventures.wallet.ui.settings.help.feedback.payment.model.PaymentFeedbackViewModel

import rx.Observable

interface PaymentFeedbackScreen : BaseFeedbackScreen {

   val paymentFeedbackViewModel: PaymentFeedbackViewModel

   fun observeMerchantName(): Observable<CharSequence>

   fun showBackConfirmDialog()

   fun discardViewModelChanges()
}
